### PR TITLE
CLI-1853: Force-add docroot/autoload_runtime.php to push:artifact scaffold files

### PR DESCRIPTION
**Pull request by amangrover90**
## Summary

Drupal 11.4 introduced `docroot/autoload_runtime.php` as a Symfony Runtime bootstrap entry point, generated by `drupal/core-composer-scaffold`. Because this file is typically listed in `docroot/.gitignore`, `git add -A` does not pick it up during `push:artifact` builds. This causes HTTP 500 errors on deployed sites because the bootstrap file is missing from the artifact.

**Changes:**
- Add `docroot/autoload_runtime.php` to the `scaffoldFiles()` hardcoded list alongside `docroot/autoload.php`, using a `file_exists` guard so it is only force-added when present (backwards compatible with Drupal < 11.4).
- Update `mockGitAddCommit()` in `PushArtifactCommandTest` to accept a `$hasAutoloadRuntime` flag that creates the file on disk (triggering the `file_exists` check) and adds the corresponding mock expectation.
- Add `testPushArtifactWithoutAutoloadRuntime` to cover the Drupal < 11.4 case (file absent, `git add -f` not called).

## Root cause

`scaffoldFiles()` at `PushArtifactCommand.php` explicitly force-adds `docroot/autoload.php` (and other core scaffold files) because `git add -A` respects `.gitignore`. The same pattern is needed for `autoload_runtime.php`, which Drupal 11.4 added as a new scaffold file that is gitignored by default.

## Test plan

- [ ] `php vendor/bin/phpunit tests/phpunit/src/Commands/Push/PushArtifactCommandTest.php` — all 10 tests pass
- [ ] Manually run `acli push:artifact` against a Drupal 11.4 site — verify `autoload_runtime.php` is committed to the artifact branch
- [ ] Run against a Drupal < 11.4 site — verify no errors and the file is not force-added

Fixes: CLI-1853

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by danepowell** [comment]
Would it be better to read `docroot/.gitignore` and force commit everything from there instead?

The contents of that gitignore mostly duplicate what's already in `docroot/core/composer.json` drupal-scaffold with the exception of autoload.php and autoload_runtime.php.

That way when some other critical file gets added in Drupal 12, we don't have to go through this exercise again.

**Comment by amangrover90** [comment]
> Would it be better to read `docroot/.gitignore` and force commit everything from there instead?
> 
> The contents of that gitignore mostly duplicate what's already in `docroot/core/composer.json` drupal-scaffold with the exception of autoload.php and autoload_runtime.php.
> 
> That way when some other critical file gets added in Drupal 12, we don't have to go through this exercise again.

It could be, but I think it'll be problematic in scenarios where someone would have added `docroot/.gitignore` to be not overridden in drupal-scaffold extras:

```
 "extra": {
        "drupal-scaffold": {
            "gitignore": false,
            },
            "locations": {
                "web-root": "docroot/"
            }
        },
```
so in that case, `docroot/.gitignore` might not be updated by scaffolding so it won't be the reliable source of truth for the files to be committed.

Also even if was, someone might have added few things to docroot that they wouldn't want to be added to the artifact. e.g. `docroot/scripts` which they would have added to the `docroot/.gitignore` and wouldn't want to add this to the artifact when running acli command locally. 

So these are few edge cases where things can get wrongly added to or missed from the deployed artifact.

File: tests/phpunit/src/Commands/Push/PushArtifactCommandTest.php [hunk 1]
```diff
@@ -195,6 +195,30 @@ public function testPushArtifactNoPush(): void
         $this->assertStringNotContainsString('Pushing changes to Acquia Git (site@svn-3.hosted.acquia-sites.com:site.git)', $output);
     }
 
+    public function testPushArtifactWithoutAutoloadRuntime(): void
+    {
+        $applications = $this->mockRequest('getApplications');
+        $this->mockRequest('getApplicationByUuid', $applications[0]->uuid);
+        $environments = $this->mockRequest('getApplicationEnvironments', $applications[0]->uuid);
+        $localMachineHelper = $this->mockLocalMachineHelper();
+        $this->setUpPushArtifact($localMachineHelper, $environments[0]->vcs->path, [$environments[0]->vcs->url], 'master:master', true, true, false, true, false);
```
[the remark this answers is not in the document]
Comment (deepakmishra2) [reply tests/phpunit/src/Commands/Push/PushArtifactCommandTest.php:204]: The setUpPushArtifact() call in the new test has 9 positional args, 5 of them booleans. It's unreadable:
can we do it this way
```
$this->setUpPushArtifact(
    localMachineHelper: $localMachineHelper,
    vcsPath: $environments[0]->vcs->path,
    vcsUrls: [$environments[0]->vcs->url],
    destGitRef: 'master:master',
    push: false,
    hasAutoloadRuntime: false,
);
```

File: tests/phpunit/src/Commands/Push/PushArtifactCommandTest.php [hunk 2]
```diff
@@ -363,6 +387,22 @@ protected function mockGitAddCommit(ObjectProphecy $localMachineHelper, string $
             'docroot/autoload.php',
         ], null, $artifactDir, false)
             ->willReturn($process->reveal())->shouldBeCalled();
+        $runtimeFile = Path::join($artifactDir, 'docroot', 'autoload_runtime.php');
+        if ($hasAutoloadRuntime) {
+            // Simulate Drupal 11.4+ where autoload_runtime.php is generated.
+            @mkdir(Path::join($artifactDir, 'docroot'), 0777, true);
```
[the remark this answers is not in the document]
Comment (deepakmishra2) [reply tests/phpunit/src/Commands/Push/PushArtifactCommandTest.php:393]: I think we should do this

**Comment by deepakmishra2** [comment]
@danepowell - Can you review this and please approve once looks good.

File: src/Command/Push/PushArtifactCommand.php [hunk 3]
```diff
@@ -457,6 +457,13 @@ private function scaffoldFiles(string $artifactDir): array
             }
         }
         $this->scaffoldFiles[] = 'docroot/autoload.php';
+        // autoload_runtime.php was introduced in Drupal 11.4 as the Symfony
+        // Runtime bootstrap entry point. It is not in core's file-mapping so
+        // it must be force-added like autoload.php, but only when present —
+        // Drupal < 11.4 does not generate this file.
+        if (file_exists(Path::join($artifactDir, 'docroot', 'autoload_runtime.php'))) {
```
Comment (vishalkhode1) [inline src/Command/Push/PushArtifactCommand.php:464]: I think then we've do the same for Code Studio as well ?
cc: @kalindi-adhiya  / @paras-raba  / @akashkska

**Review by vishalkhode1 — DISMISSED** [review]

File: src/Command/Push/PushArtifactCommand.php (hunk 3 above)
Comment (paras-raba) [reply src/Command/Push/PushArtifactCommand.php:464]: Yep:

```
function detect_scaffold_files() {
  echo "Detecting scaffold files..."
  local core_composer="$DIR/docroot/core/composer.json"
  SCAFFOLD_FILES=("docroot/.htaccess" "docroot/robots.txt" "docroot/autoload.php")

  if [ -f "$core_composer" ]; then
    if command -v jq &> /dev/null; then
      while IFS= read -r scaffold_file; do
        if [ -n "$scaffold_file" ]; then
          scaffold_file="${scaffold_file//\[web-root\]/docroot}"
          if [[ ! " ${SCAFFOLD_FILES[@]} " =~ " ${scaffold_file} " ]]; then
            SCAFFOLD_FILES+=("$scaffold_file")
          fi
        fi
      done < <(jq -r '.extra["drupal-scaffold"]["file-mapping"] | keys[]' "$core_composer" 2>/dev/null)
    else
      echo "Warning: jq is not installed. Using default scaffold files."
    fi
  else
    echo "Core composer.json not found at $core_composer. Using default scaffold files."
  fi

  echo "Detected scaffold files: ${SCAFFOLD_FILES[*]}"
}
```

Comment (deepakmishra2) [reply src/Command/Push/PushArtifactCommand.php:464]: @vishalkhode1 Should we have a separate ticket or handle in this ticket? since this is related to code studio

Comment (vishalkhode1) [reply src/Command/Push/PushArtifactCommand.php:464]: As discussed in DSM today, please create a separate ticket.

Comment (deepakmishra2) [reply src/Command/Push/PushArtifactCommand.php:464]: Done - https://acquia.atlassian.net/browse/GL-4582

**Review by deepakmishra2 — APPROVED** [review]
LGTM

**Review by vishalkhode1 — APPROVED** [review]

**Comment by deepakmishra2** [comment]
@copilot can you rerun the CI , i see codecov is running from long time
